### PR TITLE
feat: dry-run's client/server/none options with backwards compat

### PR DESCRIPTION
**Pull request by colindean**
kubectl allows options [client|server|none] instead of just a boolean, starting at some version I'm not sure of. This enables that behavior with a fallback to if the user has passed true or false so folks don't need to update their configurations.

I've mapped true->client and false->none but otherwise treating the parameter as a string.

This is the warning seen when running plugin v0.3.0:

>     W1221 21:38:34.228830      22 helpers.go:653] --dry-run=false is deprecated (boolean value) and can be replaced with --dry-run=none.

### CODE REVIEW

**Comment by colindean** [comment]
This getting kinda annoying the Vela logs:

```
WARNING: version difference between client (1.24) and server (1.21) exceeds the supported minor version skew of +/-1
$ /bin/kubectl --kubeconfig=/root/.kube/config apply --dry-run=false --filename=k8s/myproject/training_pvc.yaml
W0126 20:50:13.520375      22 helpers.go:653] --dry-run=false is deprecated (boolean value) and can be replaced with --dry-run=none.
persistentvolumeclaim/myteam--myproject--training--output unchanged
$ /bin/kubectl --kubeconfig=/root/.kube/config apply --dry-run=false --filename=k8s/anotherproject/training_pvc.yaml
W0126 20:50:14.763196      41 helpers.go:653] --dry-run=false is deprecated (boolean value) and can be replaced with --dry-run=none.
persistentvolumeclaim/myteam--anotherproject--training--output unchanged
$ /bin/kubectl --kubeconfig=/root/.kube/config apply --dry-run=false --filename=k8s/anotherproject/training_configmap.yaml
W0126 20:50:15.177479      59 helpers.go:653] --dry-run=false is deprecated (boolean value) and can be replaced with --dry-run=none.
configmap/myteam--anotherproject--training--config unchanged
$ /bin/kubectl --kubeconfig=/root/.kube/config apply --dry-run=false --filename=k8s/anotherproject/tag-and-deploy_configmap.yaml
W0126 20:50:15.588353      78 helpers.go:653] --dry-run=false is deprecated (boolean value) and can be replaced with --dry-run=none.
configmap/myteam--anotherproject--tag-and-deploy--config created
```

Can this PR get merged and released?

File: cmd/vela-kubernetes/apply_test.go [hunk 1]
```diff
@@ -37,7 +37,83 @@ func TestKubernetes_Apply_Command(t *testing.T) {
 			fmt.Sprintf("--context=%s", c.Context),
 			fmt.Sprintf("--namespace=%s", c.Namespace),
 			"apply",
-			fmt.Sprintf("--dry-run=%t", a.DryRun),
+			fmt.Sprintf("--dry-run=none"),
```
Comment (plyr4) [inline cmd/vela-kubernetes/apply_test.go:40]:
```suggestion
			"--dry-run=none",
```

File: cmd/vela-kubernetes/apply_test.go [hunk 2]
```diff
@@ -37,7 +37,83 @@ func TestKubernetes_Apply_Command(t *testing.T) {
...
+	// nolint: gosec // testing purposes
+	for _, file := range a.Files {
+		want := exec.Command(
+			_kubectl,
+			fmt.Sprintf("--kubeconfig=%s", c.Path),
+			fmt.Sprintf("--cluster=%s", c.Cluster),
+			fmt.Sprintf("--context=%s", c.Context),
+			fmt.Sprintf("--namespace=%s", c.Namespace),
+			"apply",
+			fmt.Sprintf("--dry-run=client"),
```
Comment (plyr4) [inline cmd/vela-kubernetes/apply_test.go:78]:
```suggestion
			"--dry-run=client",
```

File: cmd/vela-kubernetes/apply.go [hunk 3]
```diff
@@ -59,7 +59,19 @@ func (a *Apply) Command(c *Config, file string) *exec.Cmd {
...
-	flags = append(flags, fmt.Sprintf("--dry-run=%t", a.DryRun))
+	if len(a.DryRun) > 0 {
+		dryRunOpt := "none"
+		switch a.DryRun {
+		case "true":
+			dryRunOpt = "client"
+		case "false":
+			dryRunOpt = "none"
+		default:
+			dryRunOpt = a.DryRun
```
Comment (plyr4) [inline cmd/vela-kubernetes/apply.go:70]:
```suggestion
		dryRunOpt := a.DryRun

		switch a.DryRun {
		case "true":
			dryRunOpt = "client"
		case "false":
			dryRunOpt = "none"
```

**Review by plyr4 — CHANGES_REQUESTED** [review]

**Comment by plyr4** [comment]
hey @colindean sorry to review this so late, but it looks great to me, thank you

there are a couple tiny linter issues that can be resolved pretty easy so i left some suggestion comments. 
once we get those i can approve :)

**Comment by plyr4** [comment]
lgtm, ty! :D

**Review by wass3rw3rk — APPROVED** [review]

**Pull request merged**